blocks/cards/cards.js

```diff
@@ -1,14 +1,29 @@
 import { createOptimizedPicture } from '../../scripts/lib-franklin.js';
 
 export default function decorate(block) {
+  const cardsWithBorder = block.classList.contains('border');
+
   /* change to ul, li */
   const ul = document.createElement('ul');
   [...block.children].forEach((row) => {
     const li = document.createElement('li');
-    while (row.firstElementChild) li.append(row.firstElementChild);
-    [...li.children].forEach((div) => {
-      if (div.children.length === 1 && div.querySelector('picture')) div.className = 'cards-card-image';
-      else div.className = 'cards-card-body';
+    li.classList.add('card');
+    if (cardsWithBorder) {
+      li.classList.add('with-border');
+    }
+
+    const link = row.querySelector('a');
+    let parent = li;
+    if (link) {
+      link.textContent = '';
+      parent = link;
+      li.appendChild(link);
+    }
+
+    while (row.firstElementChild) parent.append(row.firstElementChild);
+    [...parent.children].forEach((div) => {
+      if (div.children.length === 1 && div.querySelector('picture')) div.className = 'card-image';
+      else div.className = 'card-body';
```

Comment:
With #23 the cards implementation could look something like https://github.com/andreituicu/franklin-jsx-like/blob/main/blocks/static-cards/static-cards.js 
(for dynamic cards in case we have any based on APIs https://github.com/andreituicu/franklin-jsx-like/blob/main/blocks/dynamic-cards/dynamic-cards.js )

Just to give an idea 🙂